O/OpenCL_Headers/build_tarballs.jl

```diff
@@ -1,35 +1,54 @@
 # Note that this script can accept some limited command-line arguments, run
 # `julia build_tarballs.jl --help` to see a usage message.
-using BinaryBuilder
+using BinaryBuilder, Pkg
 
 name = "OpenCL_Headers"
-version = v"2020.03.13"
+version = v"2022.9.23"
 
-# Collection of sources required to build this package
+# Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://github.com/KhronosGroup/OpenCL-Headers/archive/v2020.03.13.tar.gz",
-                  "664bbe587e5a0a00aac267f645b7c413586e7bc56dca9ff3b00037050d06f476"),
+    GitSource("https://github.com/KhronosGroup/OpenCL-Headers.git", "4c50fabe3774bad4bdda9c1ca92c82574109a74a"),
+    FileSource("https://patch-diff.githubusercontent.com/raw/KhronosGroup/OpenCL-Headers/pull/209.patch",
+               "c3afd4ad0a37f0b61c0b8656ca4914002ba7994bba05aa2c47fde59b652289c9")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd OpenCL-Headers-*
-install_license LICENSE
+cd $WORKSPACE/srcdir
 
-mkdir ${prefix}/include
-mv CL ${prefix}/include/
+install_license ./OpenCL-Headers/LICENSE
+
+patch ./OpenCL-Headers/tests/test_headers.c 209.patch
+
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -S ./OpenCL-Headers -B ./OpenCL-Headers/build
+cmake --build ./OpenCL-Headers/build --target install -j${nproc}
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = [AnyPlatform()]
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl")
+]
```

Comment:
The previous recipe was just downloading a generic tar ball, the new one actually builds the source for the limited number of platforms. The build fails in other platforms. Should I list all of them anyways?

Reply:
This is not building anything, this is just moving header files to the prefix.  All tarballs you built in https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=22722&view=logs&j=983b6c68-8f90-5831-4b74-0e190ce0b0f9 for 11 different platforms are bit-by-bit identical (you can see that by looking at the git tree hash and the sha256sum of the main tarball in the logs), because that's what it's supposed to be, and that's the entire reason why [`AnyPlatform`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.AnyPlatform) exists and it was used here in the first place.

Reply:
I am following the instructions by the OpenCL maintainers assuming that this is the best set of instructions one could possibly use to build the OpenCL-Headers project: https://github.com/KhronosGroup/OpenCL-Headers#example-build

If you prefer to just copy the raw files from a given commit manually, it may work this time, but not necessarily in the future if they start modifying the source on a per-platform basis. Should I copy the files or preserve their instructions?

Reply:
I'm only telling you that you have to restore `[AnyPlatform()]`, what you're saying here looks orthogonal to that.